src/ext/java/org/opentripplanner/ext/legacygraphqlapi/generated/LegacyGraphQLDataFetchers.java

```diff
@@ -1093,13 +1093,31 @@ public interface LegacyGraphQLServiceTimeRange {
   }
 
   public interface LegacyGraphQLStep {
+    public DataFetcher<Object> absoluteDirection();
+
+    public DataFetcher<Iterable<TransitAlert>> alerts();
+
+    public DataFetcher<Boolean> area();
+
+    public DataFetcher<Boolean> bogusName();
+
     public DataFetcher<Double> distance();
 
     public DataFetcher<Iterable<org.opentripplanner.model.plan.ElevationProfile.Step>> elevationProfile();
 
+    public DataFetcher<String> exit();
+
     public DataFetcher<Double> lat();
 
     public DataFetcher<Double> lon();
+
+    public DataFetcher<Object> relativeDirection();
```

Comment:
I'm wondering why the code generator doesn't use the generated enum type here.

@hannesj Do you know?

Reply:
You need to add it to the `src/ext/java/org/opentripplanner/ext/legacygraphqlapi/generated/graphql-codegen.yml`

Reply:
Enums elsewhere (e.g. `RouteType`) get mapped to `String`, is that the recommended way to do it?

Reply:
I think the best way is to map it to the generated type, i.e. `LegacyGraphQLRelativeDirection`. 

Take the example of the form factor.

It's configured like this: https://github.com/opentripplanner/OpenTripPlanner/blob/dbf6c5eeabf8d80bcd99652f578afdbe28137a8e/src/ext/java/org/opentripplanner/ext/legacygraphqlapi/generated/graphql-codegen.yml#L92

And then the internal type is mapped to the API type like this: https://github.com/opentripplanner/OpenTripPlanner/blob/9c9dd613489a348d2381acdcbeab8f86589154d7/src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLRentalVehicleTypeImpl.java#L13

If we do that we get a compiler error if we changed the internal type but have not type in schema.graphql.

WDYT?

Reply:
> WDYT?

👍